### PR TITLE
Change website to use Hugo academic

### PR DESCRIPTION
**Pull request by btracey**
This changes the gonum website to be based on hugo with the academic theme. This commit makes the basis for future additions to the site with the minimal tweaks to the theme. All changes were to the config.toml file and content folders, with the exception of a few changes to the theme. The first was making the gonum logo the thumbnail rather than the default, and the second was to change the header file in themes/academic/layouts/partials/header.html to add the meta tags for vanity imports. Lastly, this deletes the existing .git folder within the academic theme.

Otherwise, this moves the Intro to Gonum blog post (incorporating a few changes to make the markdown work), and adds a paper citing gonum to provide some minimal content.

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @kortschak @sbinet

It seems like the general recommendation is to have the hugo folder outside the gonum.github.io repository, but I'm not sure why. This seems easier with PRs.

**Comment by kortschak** [comment]
I make my github.io pages a submodule of my hugo repo and push to that. @vladimir-ch has his hugo directory as a subdir of his github.io repo.

Note that you can configure the academic quite extensively without changing the actual theme. I'll take a look later.

**Comment by btracey** [comment]
Does that mean we should make a gonum/hugo repository?

**Comment by kortschak** [comment]
Not sure. I would try to have a look at the web hooks that are available for autopublishing (I'm pretty sure I have seen that these exist).

Also note that I have academic as a submodule in my hugo repo. This makes maintenance simple - and why I have found out that you can change things without changing the theme.

**Comment by kortschak** [comment]
For the sake of review, if the structure you are planning to use is this, I would make this three commits:

1. Delete the old files.
2. Add the hugo structure in the path you are going to keep it in.
3. Add the generated site.

File: hugogonum/content/home/about.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+[distributions](https://godoc.org/gonum.org/v1/gonum/stat/distmv), 
+and [sampling](https://godoc.org/gonum.org/v1/gonum/stat/sampleuv); tools for
+[function differentiation](https://godoc.org/gonum.org/v1/gonum/diff/fd), 
+[integration](https://godoc.org/gonum.org/v1/gonum/integrate/quad),
+and [optimization](https://godoc.org/gonum.org/v1/gonum/optimize);
+[network](https://godoc.org/gonum.org/v1/gonum/graph) creation and analysis; and more.
+
+We encourage you to [get started](http://gonum.org/post/introtogonum/) with Go and Gonum if
+
+* You are tired of sluggish performance, and with fighting vectorization and C
```
Comment (kortschak) [inline hugogonum/content/home/about.md:53]: Suggest

```
* You are tired of sluggish performance, and fighting C and vectorization.
* You are struggling with managing programs as they grow larger.
* You struggle to re-use -- even the code you tried to make reusable.
* You would like easy access to parallel computing.
* You want code to be fully transparent, and want the ability to read the source code you use.
* You'd like a compiler to catch mistakes early, but hate fighting linker and unintelligible compile errors.
```

probably also in the article. This makes the lines generally shorter and fixes odd line breaks.

**Comment by kortschak** [comment]
The content of this looks good.

In Join Us, maybe also link to the contributing document, so there would be two contacts, the mailing list and contributing.

File: hugogonum/content/home/about.md (hunk 1 above)
Comment (btracey) [reply hugogonum/content/home/about.md:53]: Done

**Comment by btracey** [comment]
Reading some more, I see your point about being a submodule. The idea is that anyone that wants to edit the website would install hugo and the academic theme, and then would just push up the changes to public?

The two things I didn't see how to do were to change the thumbnail and to add the meta import statements.

**Comment by btracey** [comment]
This is what I had found so far with vanity imports and hugo: https://npf.io/2016/10/vanity-imports-with-hugo/ . It seems like there must be a better solution than changing the website with every new sub-package addition.

**Comment by kortschak** [comment]
I think it's work filing an issue against hugo for that.

**Comment by sbinet** [comment]
_wrt_ https://github.com/gonum/gonum.github.io/pull/6#issuecomment-313999441
the only solution I've found was to serve the hugo pages myself with a Go `net/http` server and inject a handler that would correctly interpret the `godoc` and `go-get` requests:
https://github.com/go-hep/go-hep.org/blob/master/main.go#L101

I agree that if `hugo` could automagically take care of this drudgery, it'd be really great :)
(well, the nice thing about having a Go `net/http` server was the clean and simple Let's encrypt support (together with the HTTP/2 support))

**Comment by btracey** [comment]
Sorry, I'm still confused about this, and discovered as much when trying to redo the commits.

We want to be able to build the website collaboratively (at least, I don't want to be the sole maintainer of the website). It seems to me like we need to have the actual hugo content somewhere on github. Otherwise, I don't see how people can make changes to the content and regenerate, seeing only the static site. 

Maybe I misinterpret steps 2 and 3. Is that suggestion to have a "hugogonum' or whatever subfolder, like is in this commit? I'll make the PR series along these lines.

**Pull request closed without merging**